CMakeLists.txt

```diff
@@ -289,6 +289,20 @@ if(OpenCL_FOUND)
   endif()
 endif()
 
+if(USE_HYPRE AND USE_MPI)
+  find_package(HYPRE)
+  if(HYPRE_FOUND)
+    set(HAVE_HYPRE 1)
+    list(APPEND opm-simulators_LIBRARIES HYPRE::HYPRE)
+  else()
+    message(WARNING "Hypre requested but not found. Continuing without Hypre support.")
+    set(USE_HYPRE OFF)
+  endif()
+elseif(USE_MPI)
```

Comment:
This needs to be either `USE_HYPRE` or `NOT USE_MPI`.

Reply:
aye